src/blockchain_utilities/blockchain_import.cpp

```diff
@@ -93,44 +92,6 @@ int get_db_flags_from_mode(const std::string& db_mode)
   return db_flags;
 }
 
-int parse_db_arguments(const std::string& db_arg_str, std::string& db_type, int& db_flags)
-{
-  std::vector<std::string> db_args;
-  boost::split(db_args, db_arg_str, boost::is_any_of("#"));
-  db_type = db_args.front();
-  boost::algorithm::trim(db_type);
-
-  if (db_args.size() == 1)
-  {
-    return 0;
-  }
-  else if (db_args.size() > 2)
-  {
-    std::cerr << "unrecognized database argument format: " << db_arg_str << ENDL;
-    return 1;
-  }
-
-  std::string db_arg_str2 = db_args[1];
-  boost::split(db_args, db_arg_str2, boost::is_any_of(","));
-
-  // optionally use a composite mode instead of individual flags
-  const std::unordered_set<std::string> db_modes {"safe", "fast", "fastest"};
-  std::string db_mode;
-  if (db_args.size() == 1)
-  {
-    if (db_modes.count(db_args[0]) > 0)
-    {
-      db_mode = db_args[0];
-    }
-  }
-  if (! db_mode.empty())
-  {
-    db_flags = get_db_flags_from_mode(db_mode);
-  }
-  return 0;
-}
```

Comment:
Why remove this ?

Reply:
I haven't found any usage of `db_flags` except for printing it to the logs here
https://github.com/monero-project/monero/pull/5878/files/7dc14c71d0e242a3253234113027c383809b5d6c#diff-e130c0e38f072141b5e24d17ec7dfa24L756
As there is no need to parse db type (assuming we remove db switching logic), I figured I could remove this function completely

Reply:
OK, I see this is now handled by core, so this is indeed obsolete.

Reply:
The general opinion is that it can go.